### PR TITLE
Add link to a survey for new phone masking users

### PR DESCRIPTION
**Pull request by Vinnl**
# New feature description

This adds a link to a survey to gather user input on how to iterate on phone masking.

# Screenshot (if applicable)

image

# How to test

With your user in the US or Canada (locally, with your browser language set to e.g. `en_US`) and language set to English, visit the website with a user that has the phone masking plan. You should see the banner.

# Checklist

- [ ] l10n changes have been submitted to the l10n repository, if any. - N/A, English only
- [x] All acceptance criteria are met.
- [x] I've added or updated relevant docs in the docs/ directory.
- [x] All UI revisions follow the [coding standards](https://github.com/mozilla/fx-private-relay/blob/main/docs/coding-standards.md), and use Protocol tokens where applicable (see `/frontend/src/styles/tokens.scss`).
- [x] Commits in this PR are minimal and [have descriptive commit messages](https://chris.beams.io/posts/git-commit/).

### CODE REVIEW

File: frontend/src/components/layout/topmessage/PhoneSurvey.module.scss [hunk 1]
```diff
@@ -0,0 +1,47 @@
+@import "../../../styles/tokens.scss";
```
Comment (Vinnl) [inline frontend/src/components/layout/topmessage/PhoneSurvey.module.scss:1]: Note: this file is a complete copy-paste from `InterviewRecruitment.module.scss`, and `PhoneSurvey.tsx` is basically `InterviewRecruitment.tsx` with a couple of renames. I didn't have a strong preference for adjusting the existing component and changing when it was displayed, or adding a new one - I went with the latter, but happy to change to the former.

**Comment by Vinnl** [comment]
Because of desk check feedback I added 66ae4890 to only show this banner for users who have already set up their Relay number, rather than showing it right after purchasing phone masking.

**Comment by Vinnl** [comment]
Now also added 630ad7a7, to avoid an API request for users who have not purchased phone masking (or were not able to).

**Review by groovecoder — COMMENTED** [review]
question (non-blocking) Can/should we power this with the `RECRUITMENT_BANNER_LINK` environment variable? We us that to include survey recruitment links in the email headers too.

**Comment by Vinnl** [comment]
> We us that to include survey recruitment links in the email headers too.

Oh, I did not know that! Good question. There was no request here to also recruit people via the emails, and given the specific conditions in which the survey link should appear here, I think making both a code change and an env var change is probably more trouble than it's worth (?), but definitely good to be aware of.

**Comment by Vinnl** [comment]
When there's an approving review it is, as far as I'm concerned :)

**Review by lloan — APPROVED** [review]
I left a minor, non-blocking question. There aren't any issues that I can find. LGTM.

**Review by maxxcrawford — COMMENTED** [review]
@lloan already reviewed, but I had this pending comment! 🙌

File: frontend/src/components/layout/topmessage/PhoneSurvey.module.scss [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+  .dismiss-button {
+    position: absolute;
+    top: 50%;
+    right: $spacing-md;
+    background-color: transparent;
+    border-style: none;
+    color: $color-white;
+    width: $dismissButtonWidth;
+    height: $dismissButtonWidth;
+    transform: translateY(-50%);
```
[the remark this answers is not in the document]
Comment (Vinnl) [reply frontend/src/components/layout/topmessage/PhoneSurvey.module.scss]: Hahaha good eye, those indeed seem to be redundant :joy: Also removed them from the other top messages: 07aa9179.

**Pull request merged**